File: 0069_Sqrt(x)/Sqrt(x)2.cs
public class Solution {
    public int MySqrt(int x) {
        if(x<=1) return x;
        int l = 1;
        int r = x;
        while(l<=r){
            var m = l + (r-l)/2;
            if(m > x/m){
                r = m-1;
            }else{
                l = m+1;
            }
        }

        return l - 1;
    }
}

File: 0069_Sqrt(x)/Sqrt(x)_2.java
class Solution {
    public int mySqrt(int x) {
        long l = 0, r = (long)x + 1;
        while(l < r) {
            long mid = (r-l)/2 + l;
            if(mid * mid > x) {
                r = mid;
            }else{
                l = mid + 1;
            }
        }

        l = l-1;
        return (int)l;
    }
}